src/parachains/zombienet.rs

```diff
@@ -515,3 +515,419 @@ impl Source {
+		File::create(pop_file_path)?;
+
+		let zombienet = Zombienet::new(
+			cache.clone(),
+			CONFIG_FILE_PATH,
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&vec!["https://github.com/r0gue-io/pop-node".to_string()]),
+		)
+		.await?;
+
+		let missing_binaries = zombienet.missing_binaries();
+		assert_eq!(missing_binaries.len(), 0);
+
+		temp_dir.close()?;
+		Ok(())
+	}
+
+	#[tokio::test]
+	async fn test_configure() -> Result<()> {
+		//cache
+		let temp_dir = tempfile::tempdir().expect("Could not create temp dir");
+		let cache = PathBuf::from(temp_dir.path());
+
+		let mut zombienet = Zombienet::new(
+			cache.clone(),
+			CONFIG_FILE_PATH,
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&vec!["https://github.com/r0gue-io/pop-node".to_string()]),
+		)
+		.await?;
+
+		let config = zombienet.configure();
+		assert!(config.is_ok());
+
+		temp_dir.close()?;
+		Ok(())
+	}
+
+	#[tokio::test]
+	#[ignore] // It takes long time to build
+	async fn test_spawn() -> Result<()> {
+		//cache
+		let temp_dir = tempfile::tempdir().expect("Could not create temp dir");
+		let cache = PathBuf::from(temp_dir.path());
+
+		let mut zombienet = Zombienet::new(
+			cache.clone(),
+			CONFIG_FILE_PATH,
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&vec!["https://github.com/r0gue-io/pop-node".to_string()]),
+		)
+		.await?;
+		let missing_binaries = zombienet.missing_binaries();
+		for binary in missing_binaries {
+			binary.source(&cache).await?;
+		}
+
+		let spawn = zombienet.spawn().await;
+		assert!(spawn.is_ok());
+
+		temp_dir.close()?;
+		Ok(())
+	}
+
+	#[tokio::test]
+	async fn test_spawn_error_no_binaries() -> Result<()> {
+		//cache
+		let temp_dir = tempfile::tempdir().expect("Could not create temp dir");
+		let cache = PathBuf::from(temp_dir.path());
+
+		let mut zombienet = Zombienet::new(
+			cache.clone(),
+			CONFIG_FILE_PATH,
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&vec!["https://github.com/r0gue-io/pop-node".to_string()]),
+		)
+		.await?;
+
+		let spawn = zombienet.spawn().await;
+		assert!(spawn.is_err());
+
+		temp_dir.close()?;
+		Ok(())
+	}
+
+	// Source tests
+	#[tokio::test]
+	async fn test_process_url() -> Result<()> {
+		let temp_dir = tempfile::tempdir().expect("Could not create temp dir");
+		let cache = PathBuf::from(temp_dir.path());
+
+		let source = Source::Url {
+			name: "polkadot".to_string(),
+			version: TESTING_POLKADOT_VERSION.to_string(),
+			url: "https://github.com/paritytech/polkadot-sdk/releases/download/polkadot-v1.7.0/polkadot".to_string()
+		};
+		let result = source.process(&cache).await;
+		assert!(result.is_ok());
+		assert!(temp_dir.path().join(POLKADOT_BINARY).exists());
+
+		temp_dir.close()?;
+		Ok(())
+	}
+
+	#[tokio::test]
+	#[ignore] // It takes long time to build
```

Comment:
When would we want to run this test? In our CI? If so, let's make sure we add `cargo test -- --ignored` to the CI job.

Reply:
Does make sense to run this in the CI? The process build contracts and parachains.

Abhishek opened an issue about that here: https://github.com/r0gue-io/pop-cli/issues/88

Reply:
Same as above. Ignore reasons must be included within the attribute.